client/gtk/src/main.rs

```diff
@@ -59,18 +59,22 @@ impl Update for Win {
     type Msg = VertexMsg;
 
     fn model(relm: &Relm<Win>, args: VertexArgs) -> VertexModel {
+        println!(
+            "Connecting to {}",
+            args.ip.clone().unwrap_or("127.0.0.1:8080".to_string())
```

Comment:
Should extract this to avoid the duplication (and cloning)

Reply:
Wdm?

Reply:
`unwrap_or` is done on the ip twice. Could be extracted into its own variable